File: cakk-api/src/main/java/com/cakk/api/validator/OperationValidator.java
package com.cakk.api.validator;

import static java.util.Objects.*;

import java.util.EnumMap;
import java.util.List;
import java.util.Map;

import jakarta.validation.ConstraintValidator;
import jakarta.validation.ConstraintValidatorContext;

import com.cakk.api.annotation.OperationDay;
import com.cakk.common.enums.Days;
import com.cakk.core.dto.param.shop.ShopOperationParam;

public class OperationValidator implements ConstraintValidator<OperationDay, List<ShopOperationParam>> {

	@Override
	public boolean isValid(final List<ShopOperationParam> operationParams, final ConstraintValidatorContext context) {
		if (isNull(operationParams)) {
			return false;
		}

		final Map<Days, Boolean> days = new EnumMap<>(Days.class);
		for (ShopOperationParam operationParam : operationParams) {
			if (days.containsKey(operationParam.getOperationDay())) {
				return false;
			} else {
				days.put(operationParam.getOperationDay(), true);
			}
		}
		return true;
	}
}

File: cakk-api/src/main/java/com/cakk/api/validator/OperationValidator.kt
package com.cakk.api.validator

import java.util.*

import jakarta.validation.ConstraintValidator
import jakarta.validation.ConstraintValidatorContext

import com.cakk.api.annotation.OperationDay
import com.cakk.common.enums.Days
import com.cakk.core.dto.param.shop.ShopOperationParam

class OperationValidator : ConstraintValidator<OperationDay, List<ShopOperationParam>> {
	override fun isValid(operationParams: List<ShopOperationParam>?, context: ConstraintValidatorContext): Boolean {
		operationParams ?: return false

		val days: MutableMap<Days, Boolean> = EnumMap(Days::class.java)
		operationParams.forEach {
			when {
				days.containsKey(it.operationDay) -> return false
				else -> days[it.operationDay] = true
			}
		}

		return true
	}
}
